### PR TITLE
Add option `--subdirectory NAME`.

### PR DESCRIPTION
**Pull request by fpottier**
This pull request adds a command line option `--subdirectory NAME`
which allows specifying where in the Github repository `packages/`
is found. For example, in the Coq repository, it is found in `released`,
whereas in the standard OCaml repository, it is found at the root.

Thus, this allows `opam-publish` to publish Coq packages, whereas
(it seems that) it previously could not do so.

### CODE REVIEW

**Comment by fpottier** [comment]
Does anyone have any feedback on this PR?
Am I the only one who finds it painful that `opam-publish` cannot be used to publish Coq packages?
Is there a chance that this can be merged into `1.3` and ideally also into `master`?

**Comment by rjbou** [comment]
Oh, opened since a long time! Sorry for the delay
LGTM for me on the idea, just some (incoming) little remarks in code.
Master has not yet been merged with 2.0, if you can port the commit on 2.0 branch (1.3 is no more used), it can be merged quite quickly

File: src/opam_publish.ml [hunk 1]
```diff
@@ -431,20 +431,33 @@ let update_mirror repo =
...
+  else filename ^ "/"
+
+let repo_package_dir (repo_subdirectory : string option) package =
+  let repo_subdirectory : string =
+    match repo_subdirectory with
+    | None ->
+        ""
+    | Some repo_subdirectory ->
+        add_trailing_slash repo_subdirectory
+  in
```
Comment (rjbou) [inline src/opam_publish.ml]: As the string is then given in argument to `OpamFilename.Dir.of_string`, and concatened using the operator `OpamFilename.Op.(/)`, there is no need to add this trailing slash

File: src/opam_publish.ml [hunk 2]
```diff
@@ -959,6 +972,12 @@ let repo_name =
          ~doc:"Local name of the repository to use (see the $(b,repo) \
                subcommand)")
 
+let repo_subdirectory =
+  Arg.(value & opt (some string) None & info ["subdirectory"]
```
Comment (rjbou) [inline src/opam_publish.ml]: I'd rather give it more expressive name, to not be mistaken with package to be published subdirectory (archive, github, etc.), maybe `--opam-repo-subdir` or `--repo-packages-dir` (not completely relevant this one).

File: src/opam_publish.ml [hunk 3]
```diff
@@ -959,6 +972,12 @@ let repo_name =
...
+         ~docv:"NAME"
+         ~doc:"Name of the subdirectory of the GitHub repo where \
```
Comment (rjbou) [inline src/opam_publish.ml]:
```suggestion
         ~doc:"Name of the subdirectory of the GitHub opam repo where \
```

File: src/opam_publish.ml (hunk 1 above)
Comment (fpottier) [reply src/opam_publish.ml]: Simplified, thanks.

File: src/opam_publish.ml (hunk 2 above)
Comment (fpottier) [reply src/opam_publish.ml]: OK, changing to `--opam-repo-subdir`.

File: src/opam_publish.ml (hunk 3 above)
Comment (fpottier) [reply src/opam_publish.ml]: OK.

**Comment by fpottier** [comment]
Thanks Raja. I have followed your suggestions. Now looking into creating a PR for the branch 2.0.

**Comment by rjbou** [comment]
> Now looking into creating a PR for the branch 2.0.

From this [blogpost](https://github.blog/2016-08-15-change-the-base-branch-of-a-pull-request/) no need to create a new pr.

and one last comment :)

File: src/opam_publish.ml [hunk 4]
```diff
@@ -431,20 +431,27 @@ let update_mirror repo =
       git ["reset"; "origin/master"; "--hard"];
     )
 
-let repo_package_dir package =
+let add_trailing_slash filename =
+  if Filename.check_suffix filename "/"
+  then filename
+  else filename ^ "/"
```
Comment (rjbou) [inline src/opam_publish.ml]: With last modifcation, unused function

Comment (fpottier) [reply src/opam_publish.ml]: Well spotted, thanks!

**Comment by fpottier** [comment]
> From this [blogpost](https://github.blog/2016-08-15-change-the-base-branch-of-a-pull-request/) no need to create a new pr.

I have never used this -- what happens if there are conflicts when I change the base branch of the PR?

**Comment by rjbou** [comment]
No idea, I never tried it

**Comment by fpottier** [comment]
I have created a patch based on 2.0 and will be opening a distinct PR.

**Comment by rjbou** [comment]
replaced by #84

**Pull request closed without merging**